db/hints/manager.cc

```diff
@@ -891,7 +893,24 @@ bool manager::end_point_hints_manager::sender::send_one_file(const sstring& fnam
             }
 
             co_await self->flush_maybe();
-            co_await self->send_one_hint(std::move(ctx_ptr), std::move(buf), rp, secs_since_file_mod, fname);
+            
+            if (utils::get_local_injector().enter("hinted_handoff_pause_hint_replay")) {
+                // We cannot send the hint because hint replay is paused.
+                // Sleep 100ms and do the whole loop again.
+                //
+                // Jumping to the beginning of the loop makes sure that
+                // - We regularly check if we should stop - so that we won't
+                //   get stuck in shutdown.
+                // - flush_maybe() is called regularly - so that new segments
+                //   are created and we help enforce the "at most 10s worth of
+                //   hints in a segment".
+                co_await sleep(std::chrono::milliseconds(100));
```

Comment:
A thought for the future: would have been nice to have a way to wait for injection disable event, instead of arbitrarily polling after 100ms.